commands/playersonline-new.js

```diff
@@ -1,366 +1,67 @@
+const { Rcon } = require('rcon-client');
+const baseport = 34228;
+const rconpw = process.env.RCONPASS;
+
+async function oneCommand(servernum) {
+    let port_to_use = baseport + Number(servernum)
+    const rcon = await Rcon.connect({
+        host: "localhost", port: port_to_use, password: rconpw
+    })
+    let res = await rcon.send('/p o')
+    return res
+}
+
+async function allCommand(msg, internal_error) {
+    await msg.channel.send("Asked for all online players: Awaiting reply from servers...")
+    let results = [];
+    for (let i = 1; i < 9; i++) {
+        let res = oneCommand(i).catch((err)=>{internal_error(err); return})
+        results.push(res)
+    }
+    let values = await Promise.all(results)
+    for (let i = 0; i < values.length; i++) {
+        if(!values[i]){
+            await msg.channel.send(`s${i+1} did not responde`); 
+            continue;
+        }
+        await msg.channel.send(`s${i+1}: ${values[i]}`)
+    }
+}
+
 module.exports = {
     name: 'ponew',
     aka: ['playersonlinenew'],
     description: 'how many players are online?',
     guildOnly: true,
     args: true,
     usage: ` <server#>`,
-    execute(msg, args) {
-        const Rcon = require('rcon-client');
-        const rconpw = process.env.RCONPASS;
+    execute(msg, args, internal_error) {
         const author = msg.author.username; //find author
-        const rconToSend = `/p o`; //get /players online.
-        const server = args[0];
-        const baseport = 34228;
-        let snum = [`1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`];
-        let all = ['all'];
-        let rconport = Number(server) + baseport
-        let p1 = Number(baseport) + 1;
-        let p2 = Number(baseport) + 2;
-        let p3 = Number(baseport) + 3;
-        let p4 = Number(baseport) + 4;
-        let p5 = Number(baseport) + 5;
-        let p6 = Number(baseport) + 6;
-        let p7 = Number(baseport) + 7;
-        let p8 = Number(baseport) + 8;
-        const rcon = new Rcon.Rcon({
-            host: "127.0.0.1",
-            port: `${rconport}`,
-            password: `${rconpw}`
-        });
-
-
+        const server =  Number(args[0]) || args[0];
 
         if (!server) { // Checks to see if the person specified a server number, then checks to see if the server number is part of the array of the servers it could be (1-8 currently)
-            msg.channel.send('Please pick a server first just a number (1-8). \`<#> <username> <reason>\`');
+            msg.channel.send('Please pick a server first just a number (1-8). \`<#> <username> <reason>\`')
+                .catch((err)=>{internal_error(err); return})
             console.log(`Kick-Did not have server number`);
             return;
         }
-        if (snum.indexOf(server) > -1) {
+
+        if (server >= 1 && server <= 1) {
```

Comment:
Maybe I'm missing this but I thought we fixed it earlier - should be > 1 <= 8 right?

```suggestion
        if (server >= 1 && server <= 8) {
```

Reply:
this file has been deleted lol